internal/compiler/langtype.rs

```diff
@@ -57,17 +51,8 @@ pub enum Type {
     Easing,
     Brush,
     /// This is usually a model
-    Array(Box<Type>),
-    Struct {
-        fields: BTreeMap<SmolStr, Type>,
-        /// When declared in .slint as  `struct Foo := { }`, then the name is "Foo"
-        /// When there is no node, but there is a name, then it is a builtin type
-        name: Option<SmolStr>,
-        /// When declared in .slint, this is the node of the declaration.
-        node: Option<syntax_nodes::ObjectType>,
-        /// derived
-        rust_attributes: Option<Vec<SmolStr>>,
-    },
+    Array(Rc<Type>),
```

Comment:
Using Rc<Type> like this means it is now possible to have Rc Cycle in the Type.
But I guess this hopefully doesn't happen in practice.

Reply:
How else would you be using Rc? Or put differently: is this just thinking out loud on your end, or is there an actual suggestion to handle it in another way?

Reply:
If you dislike this approach, then it might also impact future changes I have planned, such as shown here: https://github.com/slint-ui/slint/commit/981302291d7fbcc9c7fa9845316f3cbd73ab6712

If `Rc` is not the right choice here (should I be using `Weak`?), then please say so before I roll this out in even more places :)

Reply:
I think Olivier is thinking out loud :). The risk of cycles is real, but hopefully unlikely. The only alternative I can think of is an approach where all types are in say a vector (or tree) and the compiler and types themselves index into that (like an ecs). But that’s a far more intrusive approach overall, as the type registry becomes a parameter to practically all functions. Would be awesome to experiment with it, but out of scope for this PR :)

For the Rc usage in the Path I suggest to consider doing it without RefCell. The one place where mutation is needed (visit_mut) could be done using Rc::make_mut. That would also be semantically correct, as opposed to RefCell, I think (but I might be missing something:)

Reply:
Noted, this index-based design sounds also somewhat like what I use in `heaptrack` to efficiently store backtraces. 

Regarding the `Rc` usage: Thanks, TIL about `Rc::make_mut`. I guess my previous code was then simply wrong and I didn't notice it yet. IIUC then my code basically was `std::shared_ptr<T>` and happily mutated things in-place, even after copying the `shared_ptr` elsewhere. With `Rc::make_mut` I get `shared_ptr<const T>` semantics back, good.

But it seems that this negates the speed gain, as the `inlining` pass uses `object_tree::visit*` methods which internally do `Expression::visit_mut`, which then copies the `Path` again... I will need to spend some more time on this

Reply:
right, this was just me thinking out loud. Not a blocker.